### PR TITLE
feat: fetch deposit state and replace f64 values

### PR DESCRIPTION
**Pull request by enigbe**
## What this PR does?
- fetches the deposit status given an `address` and the `amount` deposited
- fix: replaces floating point (f64) parameters with `Decimal`
- fix: reads btc address and amount from environment variables

### CODE REVIEW

File: notes [hunk 1]
```diff
@@ -26,7 +26,7 @@ Exchange Funder (deps: OKEX, GALOY)
 -------------------------
 Trader (deps: OKEX)
 -> HedgingNeed
-=> moves money to/from trading account ( controling collateral)
+=> moves money to/from trading account ( controlling collateral)
```
Comment (bodymindarts) [inline notes:29]: did you accidentally commit your notes? Might want to add this to your global gitignore.

File: okex-client/tests/client.rs [hunk 2]
```diff
@@ -47,7 +49,7 @@ async fn client_is_missing_header() -> anyhow::Result<()> {
 
 #[tokio::test]
 async fn transfer_funding_to_trading() -> anyhow::Result<()> {
-    let amount = 0.00001;
+    let amount = Decimal::new(1, 5);
```
Comment (bodymindarts) [inline okex-client/tests/client.rs:52]: lets use from `from_str` or `dec!` micro for creating hard coded decimals in tests. I think that's more readable than the `(1, 5)` format

**Review by bodymindarts — COMMENTED** [review]
I will rebase my rate limit PR onto this

File: okex-client/tests/client.rs (hunk 2 above)
Comment (enigbe) [reply okex-client/tests/client.rs]: I had initially used `from_str`, but thought `new` was much cleaner. I'll revert.

**Review by bodymindarts — APPROVED** [review]

**Pull request merged**